app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt

```diff
@@ -51,20 +50,25 @@ class RepositoryModule {
     @Provides
     @Singleton
     fun providesDomainResolutionLibrary(): DomainResolution {
-        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first();
-        val network = Network.valueOf(firstWord.toUpperCase());
-        return Resolution.builder()
+        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first()
+        val network = Network.valueOf(firstWord.toUpperCase())
+        return try {
+            Resolution.builder()
                 .chainId(NamingServiceType.CNS, network)
                 .infura(NamingServiceType.CNS, BuildConfig.INFURA_API_KEY)
                 .build()
+        } catch (throwable: Throwable) {
+            Timber.e(throwable, "Error initializing UnstoppableDomains")
+            DummyDomainResolution()
```

Comment:
wouldn't it lead to a crash when using this dummy?

Reply:
No. It leads to a generic error :-)

Reply:
![image](https://user-images.githubusercontent.com/246473/116552710-9d0b0400-a8f9-11eb-8c38-8da6cea96253.png)